Apps/WebClient/src/ClientApp/src/stores/delegate.ts

```diff
@@ -0,0 +1,125 @@
+import { defineStore } from "pinia";
+import { ref } from "vue";
+
+import { DataSource } from "@/constants/dataSource";
+import { DelegateInvitationStatus } from "@/constants/delegateInvitationStatus";
+import { ErrorSourceType, ErrorType } from "@/constants/errorType";
+import { container } from "@/ioc/container";
+import { SERVICE_IDENTIFIER } from "@/ioc/identifier";
+import { ResultError } from "@/models/errors";
+import { CreateDelegateInvitationRequest } from "@/models/sharing/createDelegateInvitationRequest";
+import { DelegateInvitation } from "@/models/sharing/delegateInvitation";
+import { IDelegateService, ILogger } from "@/services/interfaces";
+import { useErrorStore } from "@/stores/error";
+
+export enum DelegateInvitationDialogStep {
+    contact = 0,
+    nickname = 1,
+    dataSources = 2,
+    expiryDate = 3,
+    review = 4,
+    sharingCode = 5,
+}
+
+export interface DelegateInvitationDialogState {
+    mode: "Create" | "Modify";
+    step: DelegateInvitationDialogStep;
+    nickname?: string;
+    email?: string;
+    expiryDate?: string;
+    dataSources?: DataSource[];
+}
+
+const defaultInvitationDialogState: DelegateInvitationDialogState = {
+    mode: "Create",
+    step: DelegateInvitationDialogStep.contact,
+};
```

Comment:
I think it would be good to consider where we keep track of the dialog being closed.  I'd suggest that occur in the store as well so the view doesn't have to send instructions to the dialog directly.

Then, if there's a default state, it should be closed (null or however we want to represent that).  The state can be populated as appropriate in the actions that open the dialog.  The state should also only need to be reset to the closed state internally within the store as a result of another action (closing or cancelling the dialog).

This could be handled when the dialog itself is implemented, though.

Reply:
Alright I'll go with the following approach (it may mature as I implement but for discussion sake):

```Typescript
dialogState = null -> dialog is closed (default)
function startNewInvitation => dialogState = { mode: "Create", step: DelegateInvitationDialogStep.contact }
function modifyInvitationExpiry(invite) = { mode: "Modify", step: DelegateInvitationDialogStep.expiry }
fucntion cancelNewInvitation => dialogState = null
```